### DISCUSSION TITLE
How to make clj-kondo detect component usages in ClojureScript?

### DISCUSSION THREAD

**Discussion by mauehara** [clj-kondo/clj-kondo#2343] [Q&A] [answered]
Hi!!

I want to create a specific linter for a ClojureScript + Reagent project that throws an error if a certain UI component is used. I've followed the example [**Disrecommend usage of function or macro**](https://github.com/clj-kondo/clj-kondo/blob/master/doc/hooks.md#disrecommend-usage-of-function-or-macro
) in the documentation. However, it is not working with the small change I've made. Here are simplified code snippets to illustrate my question:

- `.clj-kondo/config.edn:`
```clojure
{:hooks {:analyze-call {org.acme.design-system/my-component org.acme.not-recommended/hook}}
 :linters {:org.acme/not-recommended {:level :error}}}
```
- `.clj-kondo/org/acme/not_recommended.clj`
```clojure
(ns org.acme.not-recommended
  (:require [clj-kondo.hooks-api :as api]))

(defn hook [{:keys [node]}]
  (let [name (str (first (:children node)))]
    (api/reg-finding!
     (assoc (meta node)
            :message (format "Please don't use %s" name)
            :type :org.acme/not-recommended))))
```
- `org.acme.design-system.core.cljs`
```clojure
(ns org.acme.design-system.core)

(defn my-component []
  [:p "Not recommended"])
```
When I lint:
```clojure
(ns app.core
  (:require [org.acme.design-system :refer [my-component]]))

[my-component] ;; It doesn't output anything
(my-component) ;; It outputs correctly
```
How can I make the linter output correctly for the first `[my-component]` occurency? Could you help me? 😄 
Thank you!

**Comment by borkdude** [comment]
@mauehara I think there's already a solution for this. Check this linter:

https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md#discouraged-var

**Reply by mauehara** [reply]
@borkdude Thank you for the answer. Is it possible to extend that linter to my project specifics? For instace, discourage the usage of a component **only** if a certain attribute is not set?

**Reply by borkdude** [reply]
Unfortunately not. But I guess you could write an external tool and use the clj-kondo analysis output:

https://github.com/clj-kondo/clj-kondo/blob/master/analysis/README.md

to find the reference to the var and then inspect those locations using rewrite-clj.

Or, maybe use grasp:

https://github.com/borkdude/grasp

**Reply by mauehara** [reply]
Thanks! I'll take a look on these resources.

and just to confirm: hooks in clj-kondo works for `(my-component)` and not for `[my-component]`. I am not missing anything here, right?

**Reply by borkdude** [reply, marked as answer]
Hooks work on function or macro calls (originally they were designed for macro calls). A vector isn't a function or macro call.
